### PR TITLE
refactor: Simplify `build_microvm_from_requests`

### PR DESCRIPTION
**Pull request by JonathanWoollett-Light**
## Changes

Simplifies `build_microvm_from_requests` by  removing generic `Fn` arguments which where not needed.

## Reason

Generic function arguments are unweidly and add uneeded complexity.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.
- [x] New `TODO`s link to an issue.
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by ShadowCurse — APPROVED** [review]

File: src/vmm/src/rpc_interface.rs [hunk 1]
```diff
@@ -334,20 +339,17 @@ impl<'a> PrebootApiController<'a> {
     ///
```
Comment (sudanl0) [inline src/vmm/src/rpc_interface.rs:339]: could you update the comments : 
"It takes two closures `recv_req` and `respond`"

File: src/vmm/src/rpc_interface.rs [hunk 2]
```diff
@@ -1827,60 +1839,6 @@ mod tests {
         );
     }
 
-    #[test]
-    fn test_build_microvm_from_requests() {
```
Comment (sudanl0) [inline src/vmm/src/rpc_interface.rs:1831]: can we modify the test to adapt to the new changes instead of removing it completely?

Comment (JonathanWoollett-Light) [reply src/vmm/src/rpc_interface.rs:1831]: I can't see a way to do it, the only way would be to just test a function further up that then calls this (which we are already doing). I think originally this design existed for this test, but as coverage shows this actually adds more uncovered lines.

File: src/vmm/src/rpc_interface.rs (hunk 1 above)
Comment (JonathanWoollett-Light) [reply src/vmm/src/rpc_interface.rs:337]: Removed this now redundant comment.

**Review by ShadowCurse — APPROVED** [review]

**Pull request merged**